### ISSUE TITLE
Login does not work on iOS

### ISSUE THREAD

**Issue by agarciaiglesias** [forcedotcom/SalesforceMobileSDK-CordovaPlugin#461]
Version: Latest with Salesforce SDK 7.1.2.

I am developing an hybrid application with Cordova + Angular.

The issue is that when I install the application on iOS devices the method that return the user information is not working.

I use the method "force.login()" (from force.js file) but the code does not never back to the susscce or error calbacks. (The same method is working propertly on Andorid).

I have debugged the code of the plugin and I found that the plugin is trying to get the credentials with the method "getAuthCredentials" but it never gets the credentials. 

The code never go out of the function "pokeNative()" of cordova.js and enter in an infinity loop.

**Comment by wmathurin** [comment]
Did you generate your application using [forcehybrid](https://www.npmjs.com/package/forcehybrid)?
The generated app demonstrate how to login using `force.login(...)`. For the html source code see [here](https://github.com/forcedotcom/SalesforceMobileSDK-Templates/blob/master/HybridLocalTemplate/index.html).

**Comment by agarciaiglesias** [comment]
Thank you for your quickly response.

Yes, I used forcehybrid to generate the app.

I have tried to use the same code of the example and it has the same error.

This is my code:

```
forceLogin(): Promise<any> {

        return new Promise((s, r) => {

            window.force.login(

                (data) => {

                    s(data);

                },

                (err) => {

                    r(err);

                });
        });
    }
```

**Comment by wmathurin** [comment]
Have you tried debugging (see [here](https://medium.com/@mattcroak718/debugging-your-iphone-mobile-web-app-using-safari-development-tools-71240657c487) for how to use Safari Dev tools)?
There should be errors logged to the console. Also, you could step through with the debugger and see how far it gets in https://github.com/forcedotcom/SalesforceMobileSDK-Shared/blob/master/libs/force.js#L331.

**Comment by agarciaiglesias** [comment]
yes I have debugged the code of the plugin and I found that the plugin is trying to get the credentials with the method "**getAuthCredentials**" but it never gets the credentials. The code never go out of the function "**pokeNative()**" of cordova.js and enter in an infinity loop.

There is no errors and the errorCalback of login method is never asked.

**Comment by wmathurin** [comment]
It sounds like a cordova/angular issue. Did you try posting your issue on their github? I don't think it's a Mobile SDK issue.

**Comment by wmathurin** [comment]
Is it a hybrid local or a hybrid remote application?

**Comment by wmathurin** [comment]
We are able to reproduce the issue. It's a regression. It affects hybrid remote applications only. 
Until we come up with a fix, you could either:
- use an hybrid local application instead (bundle all html/css/js in the application instead of serving them from the server)
- downgrade to Mobile SDK 7.0 (simply run `npm i -g forcehybrid@7.0`).
Thank you for reporting the issue.

**Comment by wmathurin** [comment]
Here is what's going on.

In 7.1, we are using a cordova plugin that allows us to consume dependency through cocoapod. However in the generated xcode project ends up with both the cordova framework and the cordova static library.

The compiler does not complain, but it causes issues at runtime.

We did find that issue before releasing 7.1. We made a change to forcehybrid to get rid of the libCordova.a in the xcode project file (see https://github.com/forcedotcom/SalesforceMobileSDK-Package/pull/161). It looks like it's no longer working as expected.

Fortunately, there is a simple fix.
In Xcode, click on your project, then click on the application target, then go to build phases, expand the Link Binary with Libraries. You should see libCordova.a. Remove it.

<img width="862" alt="Screenshot 2019-07-08 20 07 52" src="https://user-images.githubusercontent.com/1012459/60856029-12589280-a1c3-11e9-8e0f-82d7bbe4ab61.png">

Rebuild and restart your application. It should now work as expected.

**Comment by vivekweb** [comment]
Version: Latest with Salesforce SDK 7.1.2

I am developing application with Ionic. I had same issue and did removing from Libraries as described then able to build & run in simulators

> In 7.1, we are using a cordova plugin that allows us to consume dependency through cocoapod. However in the generated xcode project ends up with both the cordova framework and the cordova static library.

Now I'm getting issues in CLI `ionic cordova build ios` and even in `ionic cordova build ios -- --buildFlag="-UseModernBuildSystem=0"`

So I tried setting up the environment in forcehybrid and tried `cordova build ios`. Still getting same error message (given below)

**Ionic:**
The following build commands failed:
	CompileC /Users//Library/Developer/Xcode/DerivedData/-ceomlnolfibbkoexifphhftqtrqm/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/Cordova.build/Objects-normal/x86_64/CDVViewController.o Cordova/CordovaLib/Classes/Public/CDVViewController.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler

**Forcehybrid:**
The following build commands failed:
	CompileC /Users/vkrishnan016/Library/Developer/Xcode/DerivedData/blank1-gbtnailyrbqfncgtwjyxciisqgcm/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/Cordova.build/Objects-normal/x86_64/CDVViewController.o Cordova/CordovaLib/Classes/Public/CDVViewController.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler

Is there any workaround or fixes available?

**Comment by agarciaiglesias** [comment]
> Here is what's going on.
> 
> In 7.1, we are using a cordova plugin that allows us to consume dependency through cocoapod. However in the generated xcode project ends up with both the cordova framework and the cordova static library.
> 
> The compiler does not complain, but it causes issues at runtime.
> 
> We did find that issue before releasing 7.1. We made a change to forcehybrid to get rid of the libCordova.a in the xcode project file (see [forcedotcom/SalesforceMobileSDK-Package#161](https://github.com/forcedotcom/SalesforceMobileSDK-Package/pull/161)). It looks like it's no longer working as expected.
> 
> **Fortunately, there is a simple fix.**
> In Xcode, click on your project, then click on the application target, then go to build phases, expand the Link Binary with Libraries. You should see libCordova.a. Remove it.
> 
> <img alt="Screenshot 2019-07-08 20 07 52" width="862" src="https://user-images.githubusercontent.com/1012459/60856029-12589280-a1c3-11e9-8e0f-82d7bbe4ab61.png">
> 
> Rebuild and restart your application. It should now work as expected.

Thank you,

This solution is working for me!

It is going to be a new version with this bug solved?

Thanks again.

**Comment by wmathurin** [comment]
forcehybrid is supposed to create the xcode project without the libCordova.a - but it doesn't seem to do it in all cases. We want to fix that bug in our next release (the soon to be released 7.2).
